### ISSUE TITLE
[Request] Add mpv as a Youtube Frontend.

### ISSUE THREAD

**Issue by kevinlekiller** [libredirect/libredirect#81]
[mpv has the ability to play Youtube videos](https://mpv.io/manual/master/#options-no-ytdl).

Examples:

`mpv https://youtu.be/aqz-KE-bpKQ`

[Using the ytdl protocol](https://github.com/mpv-player/mpv/blob/master/DOCS/man/mpv.rst#protocols): `mpv ytdl://aqz-KE-bpKQ`

It also opens links from invidious / piped, although proxying through them isn't functional, seems to be a limitation of yt-dlp?

**Comment by ManeraKai** [comment]
![image](https://user-images.githubusercontent.com/40805353/156867967-bd0cd7a7-c405-4970-af3c-8cb215744f9b.png)

**Comment by kevinlekiller** [comment]
Odd, could be your version of youtube-dl (or yt-dlp if you're using that instead) is outdated?

You can check with `--version`, latest version of youtube-dl is 2021.12.17

Do you get any error from `youtube-dl https://youtu.be/aqz-KE-bpKQ` ?

**Comment by ManeraKai** [comment]
my youtube-dl is really old: `2020.03.24`

**Comment by kevinlekiller** [comment]
Ah, that's probably why, when the version is too old, it can't fetch the video unfortunately.

**Comment by ManeraKai** [comment]
I want to now know how to add a custom protocol to Firefox to open that command like what FreeTube and Yattee have. It's weird that mpv doesn't have a one.

**Comment by kevinlekiller** [comment]
I think it needs a .desktop file with the correct mime information, perhaps mpv from flatpak installs that desktop file, will check.

**Comment by kevinlekiller** [comment]
Freetube has `MimeType=x-scheme-handler/freetube;`.

mpv doesn't have something like that: https://github.com/mpv-player/mpv/blob/master/etc/mpv.desktop

It was rejected : https://github.com/mpv-player/mpv/issues/7985

Might be better to close this issue until (if) mpv decides to add that feature?

**Comment by ManeraKai** [comment]
I'll reopen this issue if there's any progress on their side.

**Comment by 0xlua** [comment]
As of version v0.40.0 mpv supports `x-scheme-handler/mpv`: https://github.com/mpv-player/mpv/pull/15898

tl;dr: prefix `https://youtube.com/watch...` with `mpv://`, though at least on linux you have to add an extra association for `x-scheme-handler/mpv` -> `mpv.desktop`
